PLIP 10902: New collections
===========================

PLIP ticket: http://dev.plone.org/plone/ticket/10902

Review by Craig Haynal (cah190@psu.edu, cah190 on irc)

The PLIP was reviewed on Mac OS X 10.6.8 using python 2.6.6 and Safari
5.0.5.


Review steps
------------

- Ran the plip10902-new-collections.cfg

- Ran tests for plone.app.contentlisting, plone.app.collection,
plone.app.querystring, and archetypes.querywidget.

- Created and examined a new Plone site to look for any issues.

- Examined code.


Notes and observations
----------------------

- All tests in plone.app.contentlisting pass.

- 5 errors in the plone.app.collection tests: test_getFoldersAndImages,
test_limitNumber, test_searchResults, test_viewingCollection,
testPortlet.

- 5 errors in the plone.app.querystring tests: testGettingConfiguration,
testQueryBuilderHTML, testQueryBuilderNumberOfResults, 
testQueryBuilderNumberOfResultsView, testQueryBuilderQuery

- archetypes.querywidget has 1 failure in doctest integration.txt

- Attempting to install plone.app.collection results in:

 (innermost last):

    * Module ZPublisher.Publish, line 126, in publish
    * Module ZPublisher.mapply, line 77, in mapply
    * Module ZPublisher.Publish, line 46, in call_object
    * Module Products.CMFQuickInstallerTool.QuickInstallerTool, line 575, in installProducts
    * Module Products.CMFQuickInstallerTool.QuickInstallerTool, line 512, in installProduct
      __traceback_info__: ('plone.app.collection',)
    * Module Products.GenericSetup.tool, line 323, in runAllImportStepsFromProfile
      __traceback_info__: profile-plone.app.collection:default
    * Module Products.GenericSetup.tool, line 1084, in _runImportStepsFromContext
    * Module Products.GenericSetup.tool, line 998, in _doRunImportStep
      __traceback_info__: typeinfo
    * Module Products.CMFCore.exportimport.typeinfo, line 224, in importTypesTool
    * Module Products.GenericSetup.utils, line 831, in importObjects
      __traceback_info__: portal_types
    * Module Products.GenericSetup.utils, line 507, in _importBody
    * Module Products.CMFCore.exportimport.typeinfo, line 209, in _importNode
    * Module Products.GenericSetup.utils, line 565, in _initObjects
      __traceback_info__: ('Collection', 'Factory-based Type Information with dynamic views')
    * Module OFS.ObjectManager, line 323, in _setObject
    * Module OFS.ObjectManager, line 119, in checkValidId

BadRequest: The id "Collection" is reserved. 

- Unable to test new collections through the Plone.


Conclusion
----------

This PLIP seems to be broken at the moment.  The test errors and failures will need to be addressed
and it must be installable (or installed by default) to be usable.  Not ready for merging.
